lib/portlayer/storage/nfs/volume_test.go

```diff
@@ -0,0 +1,516 @@
+	testPath := path.Join(os.TempDir(), nfsTestDir)
+	os.Mkdir(testPath, 0755)
+	result := m.Run()
+	os.RemoveAll(testPath)
+	os.Exit(result)
+}
+
+func TestSimpleVolumeStoreOperations(t *testing.T) {
+	mockMount := MockMount{}
+	testVolName := "testVolume"
+	dirpath := path.Join(os.TempDir(), nfsTestDir)
+
+	targetURL := url.URL{Path: dirpath}
+	op := trace.NewOperation(context.TODO(), "TestOp")
+
+	//Create a Volume Store
+	vs, err := NewVolumeStore(op, "testStore", &targetURL, mockMount)
+	if !assert.Nil(t, err, "Failed during call to NewVolumeStore with err (%s)", err) {
+		return
+	}
+
+	_, err = os.Stat(dirpath)
+	if !assert.Nil(t, err, "Could not find the initial volume store directory after creation of volume store. err (%s)", err) {
+		return
+	}
+
+	if !assert.NotNil(t, vs, "Volume Store created with nil err, but return is also nil") {
+		return
+	}
+
+	info := make(map[string][]byte)
+	testInfoKey := "junk"
+	info[testInfoKey] = make([]byte, 20)
+
+	file, err := os.Open(dirpath)
+	contents, err := file.Readdir(0)
+	t.Logf("%s\n", contents)
+
+	//Create a Volume
+	vol, err := vs.VolumeCreate(op, testVolName, vs.Target, 0 /*we do not use this*/, info)
+
+	if !assert.Nil(t, err, "Failed during call to VolumeCreate with err (%s)", err) {
+		return
+	}
+
+	if !assert.Equal(t, testVolName, vol.ID, "expected volume ID (%s) got ID (%s)", testVolName, vol.ID) {
+		return
+	}
+
+	_, ok := vol.Info[testInfoKey]
+	if !assert.True(t, ok, "TestInfoKey did not exist in the return metadata map") {
+		return
+	}
+
+	//Check Metadata Pathing
+	metadataPath := path.Join(dirpath, metadataDir)
+	volumePath := path.Join(dirpath, VolumesDir)
+
+	metaFilesDir, err := os.Open(metadataPath)
+	if !assert.Nil(t, err, "opening the metadata directory failed with err (%s)", err) {
+		return
+	}
+
+	volumeDir, err := os.Open(volumePath)
+	if !assert.Nil(t, err, "Opening the volume directory failed with err (%s)", err) {
+		return
+	}
+
+	metaDirEntries, err := metaFilesDir.Readdir(0)
+	if !assert.Nil(t, err, "Failed to read the metadata directory with err (%s)", err) {
+		return
+	}
+
+	volumeDirEntries, err := volumeDir.Readdir(0)
+	if !assert.Nil(t, err, "Failed to read the volume data directory with err (%s)", err) {
+		return
+	}
+
+	t.Logf("meta directory (%s)", metaDirEntries)
+	if !assert.Equal(t, len(metaDirEntries), 1, "expected metadata directory to have 1 entry and it had (%s)", len(metaDirEntries)) {
+		return
+	}
+	t.Logf("vol directory (%s)", volumeDirEntries)
+	if !assert.Equal(t, len(volumeDirEntries), 1, "expected metadata directory to have 1 entry and it had (%s)", len(volumeDirEntries)) {
+		return
+	}
+
+	metaFilesDir.Close()
+	volumeDir.Close()
+
+	//Remove the same Volume
+	err = vs.VolumeDestroy(op, vol)
+	if !assert.Nil(t, err, "Failed during a call to VolumeDestroy with err (%s)", err) {
+		return
+	}
+
+	metaFilesDir, err = os.Open(metadataPath)
+	if !assert.Nil(t, err, "opening the metadata directory failed with err (%s)", err) {
```

Comment:
I do need to check to make sure that the path to the metadata file for the specific volume is gone though. I will add that. the path we need to check for should be(in this case) `/tmp/testStore/metaData/volOne` all of the contents of the map from volOne will be stored in the `volOne` directory.